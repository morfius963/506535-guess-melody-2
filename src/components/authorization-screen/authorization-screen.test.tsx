import * as React from "react";
import * as renderer from "react-test-renderer";
import AuthorizationScreen from "./authorization-screen";

describe(`snapshot test`, () => {
  it(`Component correctly renders`, () => {
    const props = {
      formSubmitHandler: jest.fn(),
      userInputHandler: jest.fn()
    };

    const tree = renderer
      .create(<AuthorizationScreen {...props} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
