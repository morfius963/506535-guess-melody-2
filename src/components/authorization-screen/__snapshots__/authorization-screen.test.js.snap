// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test Component correctly renders 1`] = `
<section
  className="login"
>
  <div
    className="login__logo"
  >
    <img
      alt="Угадай мелодию"
      height="83"
      src="img/melody-logo.png"
      width="186"
    />
  </div>
  <h2
    className="login__title"
  >
    Необходима авторизация
  </h2>
  <p
    className="login__text"
  >
    Представтесь!
  </p>
  <form
    action=""
    className="login__form"
    onSubmit={[MockFunction]}
  >
    <p
      className="login__field"
    >
      <label
        className="login__label"
        htmlFor="email"
      >
        Логин
      </label>
      <input
        className="login__input"
        id="email"
        name="email"
        onChange={[MockFunction]}
        required={true}
        type="text"
      />
    </p>
    <p
      className="login__field"
    >
      <label
        className="login__label"
        htmlFor="password"
      >
        Пароль
      </label>
      <input
        className="login__input"
        id="password"
        name="password"
        onChange={[MockFunction]}
        required={true}
        type="password"
      />
      <span
        className="login__error"
      >
        Неверный пароль
      </span>
    </p>
    <button
      className="login__button button"
      style={
        Object {
          "zIndex": "1",
        }
      }
      type="submit"
    >
      Войти
    </button>
  </form>
</section>
`;
